mlir/test/CMakeLists.txt

```diff
@@ -45,6 +46,14 @@ if(ROCMLIR_DRIVER_XDLOPS_TEST_ENABLED)
   set(MLIR_XDLOPS "-x2")
 endif()
 
+# Passed to lit.site.cfg.py.in to set the -no-mfma flag for the mlir E2E tests.
+set(ROCMLIR_DISABLE_ADVANCED_FEATURE "")
+if(ROCMLIR_DRIVER_DISABLE_MFMA)
```

Comment:
Having stared at this, we might want to make this more general? Instead of having `_DISABLE_BLAH`, we could have `ROCMLIR_DRIVER_FORCE_BLAH` , which is by default unset. If you set it to ON, you get a `--blah` in your options (ex. `--mfma`), and setting it to OFF gives you `--no-blah`)

Then, in the drivers, you check to see if the boolean option has been set: if the uses passed neither `--blah` nor `--no-blah`, use the system default)